lib/jsonpath/parser.rb

```diff
@@ -102,6 +106,31 @@ def parse_exp(exp)
 
     private
 
+    # /foo/i -> Regex.new("foo", Regexp::IGNORECASE) without using eval
+    # also supports %r{foo}i
```

Comment:
@grosser Would you mind adding some test case for these options please to see if they are even usable and don't break things unexpectedly?